File: ruby/question_add.rb
require 'date'
require 'faraday'

#
# Add a Question to a Branch/Site
#

# Required API Variables
require './vars.rb'

# URL and Path
url = "https://#{DOMAIN}.gimlet.us"
path = "/api/v1/sites/#{SITE_ID}/questions"

date_time = DateTime.now

# Field values
fields = {}
fields["site_id"] = SITE_ID
fields["question"]= {}
fields["question"]["location_name"] = "Reference"
fields["question"]["duration_name"] = "0-5 minutes"
fields["question"]["format_name"]   = "Phone"
fields["question"]["questioner_name"] = "Staff"
fields["question"]["question_type_name"] = "Technology"
fields["question"]["difficulty_name"] = "1"
fields["question"]["tag_list"] = "example_tag1 example_tag2"
fields["question"]["question"] = "Test at #{Time.now.strftime("%I:%M %p")}!"
fields["question"]["answer"] = "Another test!"
fields["question"]["initials"] = "test"
fields["question"]["asked_at_time"] = date_time.strftime("%I:%M %p")
fields["question"]["asked_at_date"] = date_time.strftime("%m/%d/%Y")

# New connection with base URL and path
conn = Faraday.new(url: url)        # New connection with base URL
conn.basic_auth(EMAIL, API_KEY)     # Set the authentication header
response = conn.post(path, fields)  # POST API request

# Print the results
results = response.body
puts "RESULTS: #{results.inspect}"

File: php/question_add.php
<?php
require './vars.php';

$api_url = "https://{$DOMAIN}.gimlet.us/api/v1/sites/{$SITE_ID}/questions";

$now_time = date('G:i');
$now_date = date('m/d/Y');

// Change these to be what you actually need!
$fields = [
    "site_id" => $SITE_ID,
    "question[location_name]" => "Reference",
    "question[duration_name]" => "0-5 minutes",
    "question[format_name]" => "Phone",
    "question[questioner_name]" => "Staff",
    "question[question_type_name]" => "Technology",
    "question[difficulty_name]" => "1",
    "question[tag_list]" => "example_tag1 example_tag2",
    "question[question]" => "Test at {$now_time}!",
    "question[answer]" => "Another test!",
    "question[initials]" => "test",
    "question[asked_at_time]" => $now_time,
    "question[asked_at_date]" => $now_date,
];

$req = curl_init($api_url);
$credentials = "{$EMAIL}:{$API_KEY}";
curl_setopt($req, CURLOPT_USERPWD, $credentials);
curl_setopt($req, CURLOPT_POST, true);
curl_setopt($req, CURLOPT_POSTFIELDS, $fields);
curl_setopt($req, CURLOPT_RETURNTRANSFER, true);

$data = curl_exec($req);
$response_code = curl_getinfo($req, CURLINFO_RESPONSE_CODE);
// Do what you need with the data
echo("{$response_code} {$data}");

// ?>
